ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KtlintSuppressionRuleTest.kt

```diff
@@ -0,0 +1,798 @@
+                ).isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a SuppressWarnings annotation on a declaration`() {
+            val code =
+                """
+                @SuppressWarnings("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar")
+                val foo = "foo"
+                """.trimIndent()
+            val formattedCode =
+                """
+                @SuppressWarnings("ktlint:standard:bar", "ktlint:standard:foo", "ktlint:standard:foo-bar")
+                val foo = "foo"
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolations(
+                    LintViolation(1, 27, "Identifier to suppress ktlint rule must be fully qualified with the rule set id"),
+                    LintViolation(1, 64, "Identifier to suppress ktlint rule must be fully qualified with the rule set id"),
+                ).isFormattedAs(formattedCode)
+        }
+    }
+
+    @Test
+    fun `Given an EOL comment with a ktlint-disable directive not preceded by code leaf on same line`() {
+        val code =
+            """
+            val foo = "foo"
+            // ktlint-disable
+            val bar = "bar"
+            """.trimIndent()
+        val formattedCode =
+            """
+            val foo = "foo"
+            val bar = "bar"
+            """.trimIndent()
+        ktlintSuppressionRuleAssertThat(code)
+            .hasLintViolation(2, 4, "Directive 'ktlint-disable' in EOL comment is ignored as it is not preceded by a code element")
+            .isFormattedAs(formattedCode)
+    }
+
+    @Nested
+    inner class `Given an import statement` {
+        @Test
+        fun `Given an EOL comment with a ktlint-disable directive on an import`() {
+            val code =
+                """
+                import foo.bar
+                import foobar.* // ktlint-disable no-wildcard-imports
+                """.trimIndent()
+            val formattedCode =
+                """
+                @file:Suppress("ktlint:standard:no-wildcard-imports")
+
+                import foo.bar
+                import foobar.*
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(2, 20, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given an EOL comment with a ktlint-disable directive on an import and an existing @file Suppress annotation`() {
+            val code =
+                """
+                @file:Suppress("aaa", "zzz")
+
+                import foo.bar
+                import foobar.* // ktlint-disable no-wildcard-imports
+                """.trimIndent()
+            val formattedCode =
+                """
+                @file:Suppress("aaa", "ktlint:standard:no-wildcard-imports", "zzz")
```

Comment:
do we have a wrapping rule around annotation? If yes then we should also add one TC which after autocorrect violates the wrapping rule

Reply:
Annotations are wrapped similar to any function call. The wrapping rule is ran later in the process. I see no need to test the combination explicitly.